kpops/component_handlers/kafka_connect/model.py

```diff
@@ -44,19 +49,47 @@ def json_schema_extra(schema: dict[str, Any], model: type[BaseModel]) -> None:
         extra="allow",
         alias_generator=to_dot,
         json_schema_extra=json_schema_extra,
+        populate_by_name=True,
     )
 
     @field_validator("connector_class")
+    @classmethod
     def connector_class_must_contain_dot(cls, connector_class: str) -> str:
         if "." not in connector_class:
             msg = f"Invalid connector class {connector_class}"
             raise ValueError(msg)
         return connector_class
 
+    @pydantic.field_validator("topics", mode="before")
+    @classmethod
+    def validate_topics(cls, topics: Any) -> list[KafkaTopic] | None | Any:
+        if isinstance(topics, str):
+            return [KafkaTopic(name=topic_name) for topic_name in topics.split(",")]
+        return topics
```

Comment:
I am wondering if this is the correct behaviour here. Shouldn't the "validate" method actually validate the input? For instance, if the user passes a malformed input. The method should check it and raise a ValidationError.

Reply:
I called it `validate_topics` based on the usage of a Pydantic "`BeforeValidator`" and the examples in their docs. Essentially it is a custom deserializer (that's what the "before" indicates). 
from their docs:
> Before validators run before Pydantic's internal parsing and validation (e.g. coercion of a str to an int). These are more flexible than After validators since they can modify the raw input, but they also have to deal with the raw input, which in theory could be any arbitrary object.

The actual validation is done during by Pydantic during the init. If you want I can name it `deserialize_topics` instead

Reply:
Yeah I like that name more. I was using this functionality in another project and at first it seemed very confusing to me!